File: data/connection/src/commonMain/kotlin/com/runvpn/app/data/connection/VpnConnectionFactory.kt
package com.runvpn.app.data.connection

import com.runvpn.app.data.connection.models.Ikev2Config
import com.runvpn.app.data.connection.models.OpenVpnConfig
import com.runvpn.app.data.connection.models.OverSocksConfig
import com.runvpn.app.data.connection.models.WireGuardConfig
import com.runvpn.app.data.connection.models.XrayVpnConfig

interface VpnConnectionFactory {
    fun createXrayVpnConnection(xrayVpnConfig: XrayVpnConfig): VpnConnection
    fun createOpenVpnConnection(openVpnConfig: OpenVpnConfig): VpnConnection
    fun createWireGuardConnection(wireGuardConfig: WireGuardConfig): VpnConnection
    fun createOverSocksConnection(overSocksConfig: OverSocksConfig): VpnConnection
    fun createIkev2Connection(ikev2Config: Ikev2Config): VpnConnection
}

File: iosApp/iosApp/Connection/vpn/VpnConnectionFactory.swift
//
//  VpnConnectionFactory.swift
//  iosApp
//
//  Created by LocalAdmin on 27.03.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import shared


class DefaultVpnConnectionFactory: VpnConnectionFactory {
    
    func createIkev2Connection(ikev2Config: Ikev2Config) -> any VpnConnection {
        return XrayVpnConnection()
    }
    
    func createOpenVpnConnection(openVpnConfig: OpenVpnConfig) -> any VpnConnection {
        return XrayVpnConnection()
    }
    
    func createOverSocksConnection(overSocksConfig: OverSocksConfig) -> any VpnConnection {
        return XrayVpnConnection()
    }
    
    func createWireGuardConnection(wireGuardConfig: WireGuardConfig) -> any VpnConnection {
        return XrayVpnConnection()
    }
    
    func createXrayVpnConnection(xrayVpnConfig: XrayVpnConfig) -> any VpnConnection {
        return XrayVpnConnection()
    }
}
